ansible_base/lib/dynamic_config/__init__.py

```diff
@@ -0,0 +1,199 @@
+from __future__ import annotations
+
+import inspect
+import os
+import sys
+from pathlib import Path
+
+from django.core.exceptions import ImproperlyConfigured
+from dynaconf import Dynaconf
+from dynaconf.constants import YAML_EXTENSIONS
+from dynaconf.loaders import env_loader
+from dynaconf.loaders.base import BaseLoader
+from dynaconf.loaders.yaml_loader import yaml
+
+from ansible_base.lib.dynamic_config.settings_logic import get_mergeable_dab_settings
+
+
+def factory(
+    name: str,  # main app name to be used to name env_switcher and envvar_prefix
+    extra_envvar_prefixes: list[str] | None = None,  # extra prefixes to be used in envvar loader
+    **options,  # options to be passed to Dynaconf
+) -> Dynaconf:
+    """Create a Dynaconf instance for a specific service.
+
+    This function creates a Dynaconf instance with the following options:
+    - env_switcher: based on the app name
+    - envvar_prefix: based on the app name
+    - root_path: based on the caller module path
+    - loaders: empty list (to be loaded manually)
+    - options: any other options passed to the function (e.g. settings_files)
+      some options are not allowed and will raise an error if passed.
+
+    The Dynaconf instance returned will do the following:
+    - Define its mode based on the env_switcher e.g. "AWX_MODE"
+    - Define its envvar prefix based on the envvar_prefix e.g. "AWX"
+    - Consider multiple envvar_prefixes if extra_envvar_prefixes is passed (e.g. "AWX,FOO,BAR")
+    - Define STANDARD_SETTINGS_FILES for /etc/ansible-automation-platform/config/
+      first load yaml files from the base path, then from the app specific path.
+    - Pass all the options to the Dynaconf instance (except the invalid ones)
+      e.g:
+        - settings_files=["defaults.py"] (this is mainly for default static settings)
+        - environments=("development", "production", "quiet", "kube") (this is for modes)
+          and allow the load of additional settings files such as `development_defaults.py`
+
+    Once the Dynaconf is instantiated (and settings are loaded), the function will
+    add the DAB settings to the Dynaconf instance, and set the `is_development_mode`
+    flag based on the current mode.
+
+    Dynaconf object is returned and the caller can perform additional operations
+    such as read settings, set additional settings, load more settings files,
+    and finally export the settings to the caller module using the `export` function.
+    """
+
+    _check_options(options)
+    prefix = name.upper()
+    name = name.lower()
+    frame = inspect.currentframe().f_back
+    caller_path = os.path.dirname(inspect.getfile(frame))
```

Comment:
Minor: I'd prefer to use explicit parameter to pass a path to the function rather then implicit stack inspection magick.

Reply:
This would require passing the `__name__` attribute

```py
DYNACONF = factory(__name__, **kwargs)
```
and at the end
```py
export(__name__, DYNACONF)
```

Internally those 2 function can take the module from `sys.modules[__name__]`

I am ok with that, I just think is a bit ugly having to pass `__name__` and I don ´t see problems with inspecting the stack to get the caller's module name on non async code.

I can change to take the __name__ if needed